File: scala/src/main/scala/io/devorbit/sparklabs/batch/sales/sql/AvgProductPrice.scala
package io.devorbit.sparklabs.batch.sales.sql

object AvgProductPrice extends App {
  SqlUtils.sql(
    """
      |SELECT category, ROUND(AVG(price)) AS avg_price
      |FROM products
      |GROUP BY category
      |ORDER BY avg_price DESC
      |""".stripMargin
  ).show(false)
}

File: python/sparklabs/batch/sales/sql/avg_product_price.py
from sparklabs.batch.sales.sql import utils


def run() -> None:
    utils.sql(
        """
        SELECT category, ROUND(AVG(price)) AS avg_price
        FROM products
        GROUP BY category
        ORDER BY avg_price DESC
        """
    ).show(truncate=False)


if __name__ == "__main__":
    run()
